infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftCheInstallationLocation.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.openshift.environment;
+
+import com.google.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * OpenShiftCheInstallationLocation checks the KUBERNETES_NAMESPACE and POD_NAMESPACE environment
+ * variables to determine what namespace Che is installed in. Users should use this class to
+ * retrieve the installation namespace name.
+ *
+ * @author Tom George
+ */
+@Singleton
+public class OpenShiftCheInstallationLocation {
+
+  private static final Logger LOG = LoggerFactory.getLogger(OpenShiftCheInstallationLocation.class);
+
+  @Inject(optional = true)
+  @Named("env.KUBERNETES_NAMESPACE")
```

Comment:
And ```javax.inject``` instead of google inject.

Reply:
@skabashnyuk I believe we need to refactor this class based on the valid comments in the review e.g. move to different location + remove completely the usage of `env.POD_NAMESPACE`. wondering if this will be ok do in a separate PR though, since we really need to have a fix in 7.12.0

Reply:
@skabashnyuk  unfortunately Guice does not support constructor injection with optional dependencies.  We need to have these dependencies be optional for now because each installation method uses a different environment variable to specify the installation location.  Perhaps when we refactor the installation methods to use the same environment variable, we can revisit this?

Reply:
what about import org.eclipse.che.commons.annotation.Nullable?

Reply:
It didn't work, that was what I was asking about yesterday

Reply:
I assume it's not working this way
```java
  @Inject
  public OpenShiftCheInstallationLocation(
      @Named("env.KUBERNETES_NAMESPACE") String kubernetesNamespace,
      @Named("env.POD_NAMESPACE") String podNamespace) {
    this.kubernetesNamespace = kubernetesNamespace;
    this.podNamespace = podNamespace;
  }
```
but should work
```java
  @Inject
  public OpenShiftCheInstallationLocation(
   @Nullable  @Named("env.KUBERNETES_NAMESPACE") String kubernetesNamespace,
    @Nullable  @Named("env.POD_NAMESPACE") String podNamespace) {
    this.kubernetesNamespace = kubernetesNamespace;
    this.podNamespace = podNamespace;
  }
```
No?

Reply:
https://github.com/eclipse/che/blob/master/infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactory.java#L56-L61

Reply:
No, neither one worked.  They both complained about not being able to find a binding for  `env.POD_NAMESPACE` when it was not defined in the container, even though it was defined as `@Nullable`

Reply:
you right. Probably the first variant without injection was good too.